compiler/rustc_hir_analysis/src/variance/mod.rs

```diff
@@ -55,6 +55,9 @@ fn variances_of(tcx: TyCtxt<'_>, item_def_id: LocalDefId) -> &[ty::Variance] {
         DefKind::OpaqueTy | DefKind::ImplTraitPlaceholder => {
             return variance_of_opaque(tcx, item_def_id);
         }
+        DefKind::AssocTy if tcx.opt_rpitit_info(item_def_id.to_def_id()).is_some() => {
```

Comment:
This is not correct for `ImplTraitInTraitData::Impl`, this should just be matching on `ImplTraitInTraitData::Trait`.

Reply:
Also, why are we even reaching this at all? We shouldn't need to compute the variances of an associated type.

Reply:
@compiler-errors interesting, why is wrong for impls?  I meant, is there a test we can build that would fail? or what would could go wrong?

Reply:
Look at the `variance_of_opaque` -- it relies on the `item_bounds` of the def-id you pass in. There are no item bounds for impl associated types.